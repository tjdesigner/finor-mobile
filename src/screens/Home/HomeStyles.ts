import { Dimensions, TextProps, TouchableOpacity } from "react-native"
import styled from "styled-components/native"

export const Container = styled.View`
  flex: 1;
  background-color: #fff;
  align-items: center;
  justify-content: center;
`
export const Title = styled.Text`
  color: #000;
  text-align: center;
  font-size: 16px;
`

interface FabButtonProps {
  screenPosition?: "left" | "right" | "center"
}

export const FabButton = styled(TouchableOpacity)<FabButtonProps>`
  width: ${({ theme }) => theme.spaces.xxxLarge};
  height: ${({ theme }) => theme.spaces.xxxLarge};
  border-radius: ${({ theme }) => theme.spaces.xxLarge};
  background-color: ${({ theme }) => theme.colors.overlay};
  color: ${({ theme }) => theme.colors.white};
  align-items: center;
  justify-content: center;
  position: absolute;
  bottom: ${({ theme }) => theme.spaces.medium};
  left: ${(props) =>
    props.screenPosition === "left"
      ? ({ theme }) => theme.spaces.large
      : props.screenPosition === "right"
      ? Dimensions.get("window").width - 100 + "px"
      : Dimensions.get("window").width / 2 - 30 + "px"};
`

interface FabButtonTextProps extends TextProps {
  color?: string
}

export const FabButtonText = styled.Text<FabButtonTextProps>`
  color: ${(props) =>
    props.color ? props.color : ({ theme }) => theme.colors.white};
`
