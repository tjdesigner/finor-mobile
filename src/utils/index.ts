import { Platform } from "react-native"
export const textCamelcaseSeparation = (name: string) => {
  return name
    .split(/(?=[A-Z])/)
    .join()
    .replace(",", " ")
}

/**
 * Format Currency
 */
export const formatCurrency = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
})

const { OS } = Platform

export const ANDROID = OS === "android"
export const IOS = OS === "ios"
