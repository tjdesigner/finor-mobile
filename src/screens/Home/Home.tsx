import React from 'react';
import { View, Text, StyleSheet, Button } from 'react-native';
import { RootStackScreenProps } from '../../@types/navigation';
import theme from '../../global/styles/theme';
import { FabButton, FabButtonText } from './HomeStyles';
import { MaterialIcons } from '@expo/vector-icons'
import { testId } from './../../../e2e/testIds'

const { fabButtonAdd, fabButtonProfile } = testId.Home

const userMock = [
  { id: 1, name: 'Tiago', age: 39 }, { id: 2, name: 'Poliana', age: 34 }
]

export function Home({ navigation, route }: RootStackScreenProps<'Home'>) {

  const handleNavigation = () => {
    navigation.navigate('AddListForm', {
      users: userMock
    })
  }

  return (
    <View style={styles.container}>
      <Text style={{ fontSize: 30, fontWeight: 'bold', letterSpacing: 1, marginBottom: 8 }}>Home</Text>
      <Text style={{ fontSize: 16, letterSpacing: 1 }}>Parametros</Text>
      <FabButton testID={fabButtonProfile} screenPosition='left' onPress={() => handleNavigation()}>
        <FabButtonText>
          <MaterialIcons name='person' size={theme.fontSizeNumber.medium} />
        </FabButtonText>
      </FabButton>
      <FabButton testID={fabButtonAdd} screenPosition='right' onPress={() => handleNavigation()}>
        <FabButtonText>
          <MaterialIcons name='add' size={theme.fontSizeNumber.medium} />
        </FabButtonText>
      </FabButton>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.primary,
    alignItems: 'center',
    justifyContent: 'center'
  },
});