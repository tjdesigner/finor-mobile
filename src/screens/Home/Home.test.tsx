import * as React from 'react';
import { fireEvent, render } from "@testing-library/react-native"

import { Home } from './Home';
import { FabButton } from './HomeStyles';
import { withTheme } from '../../utils/withTheme';
import { testId } from './../../../e2e/testIds'

const { fabButtonAdd, fabButtonProfile } = testId.Home

jest.mock("@react-navigation/native", () => ({
    createNavigatorFactory: jest.fn(),
    useFocusEffect: jest.fn(),
}))

jest.mock("@react-navigation/native-stack", () => ({
    createStackNavigator: jest.fn(),
}))

interface Navigation {
    navigate(destination: string): void;
}

interface RegistrationProps {
    navigation: Navigation;
}

jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({ navigate: jest.fn() }),
    useRoute: () => ({} as any),
}));

describe('Home', () => {
    const onPress = jest.fn()

    it('should render page with clinicalAnalysis and pathological tabs enabled', () => {
        const testRenderer = render(withTheme(<Home navigation={{} as any} route={{} as any} />))
        expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('Test click event', () => {
        const { getByTestId } = render(withTheme(<FabButton testID={fabButtonAdd} onPress={onPress} />))
        fireEvent.press(getByTestId(fabButtonAdd));
        expect(onPress).toBeCalledTimes(1)
    });
})
