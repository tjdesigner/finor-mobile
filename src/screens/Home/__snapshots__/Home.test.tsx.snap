// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render page with clinicalAnalysis and pathological tabs enabled 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#43D094",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "marginBottom": 8,
      }
    }
  >
    Home
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "letterSpacing": 1,
      }
    }
  >
    Parametros
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "borderRadius": 48,
        "bottom": 24,
        "color": "white",
        "height": 60,
        "justifyContent": "center",
        "left": 40,
        "opacity": 1,
        "position": "absolute",
        "width": 60,
      }
    }
    testID="fab_button_Profile"
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
        ]
      }
    >
      <Text />
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "borderRadius": 48,
        "bottom": 24,
        "color": "white",
        "height": 60,
        "justifyContent": "center",
        "left": 650,
        "opacity": 1,
        "position": "absolute",
        "width": 60,
      }
    }
    testID="fab_button_add"
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
        ]
      }
    >
      <Text />
    </Text>
  </View>
</View>
`;
